packages/bridge/contracts/l2/L2GovernanceRelay.sol

```diff
@@ -15,21 +15,22 @@
 
 pragma solidity ^0.6.11;
 
+import "./L2CrossDomainEnabled.sol";
+
 // Receive xchain message from L1 counterpart and execute given spell
 
-contract L2GovernanceRelay {
+contract L2GovernanceRelay is L2CrossDomainEnabled {
   address public immutable l1GovernanceRelay;
 
   constructor(address _l1GovernanceRelay) public {
     l1GovernanceRelay = _l1GovernanceRelay;
   }
 
-  function relay(address target, bytes calldata targetData) external {
-    require(msg.sender == l1GovernanceRelay, "L2GovernanceRelay/not-called-by-l1GovernanceRelay");
-    // require(); todo: require that it's a top-level call releyed from L1
```

Comment:
I don't believe this would add extra security because of the address alias, but you can call this precompile function to validate that
https://github.com/OffchainLabs/arb-os/blob/e8d9696f21f203edf4be438062df13dc6902b09b/contracts/arbos/builtin/ArbSys.sol#L57